tests/generator/base.py

```diff
@@ -72,11 +72,11 @@
 UDEV_MAC_RULE = 'SUBSYSTEM=="net", ACTION=="add", DRIVERS=="%s", ATTR{address}=="%s", NAME="%s"\n'
 UDEV_NO_MAC_RULE = 'SUBSYSTEM=="net", ACTION=="add", DRIVERS=="%s", NAME="%s"\n'
 UDEV_SRIOV_RULE = 'ACTION=="add", SUBSYSTEM=="net", ATTRS{sriov_totalvfs}=="?*", RUN+="/usr/sbin/netplan apply --sriov-only"\n'
-ND_WITHIPGW = '[Match]\nName=%s\n\n[Network]\nLinkLocalAddressing=ipv6\nAddress=%s\nAddress=%s\nGateway=%s\n\
+ND_WITHIPGW = '[Match]\nName=%s\n\n[Network]\nLinkLocalAddressing=ipv6\nAddress=%s\nAddress=%s\nGateway=%s\nGateway=%s\n\
 ConfigureWithoutCarrier=yes\n'
 NM_WG = '[connection]\nid=netplan-wg0\ntype=wireguard\ninterface-name=wg0\n\n[wireguard]\nprivate-key=%s\nlisten-port=%s\n%s\
 \n\n[ipv4]\nmethod=manual\naddress1=15.15.15.15/24\ngateway=20.20.20.21\n\n[ipv6]\nmethod=manual\naddress1=\
-2001:de:ad:be:ef:ca:fe:1/128\n'
+2001:de:ad:be:ef:ca:fe:1/128\ngateway=2001:de:ad:be:ef:ca:fe:2\n'
```

Comment:
This was because of the mandatory coverage thing. If #218 is merged beforehand, I could simply convert the manual NULL-checking to use the new macro, which would have been battle-tested elsewhere anyway, rendering these changes unnecessary.

Reply:
In a future version (probably tomorrow) I'll put a proper unit test for the multipass situation in test_common.py as suggested in the review cover letter.

Reply:
Addressed. Thanks for the heads-up, I had actually misunderstood the reasons behind the multi-pass: I thought it was needed for the definition of a Wireguard interface, which retrospectively was a bit weird :grin:.